File: ailist-core/src/main/java/me/kosik/interwalled/ailist/core/model/Configuration.java
package me.kosik.interwalled.ailist.core.model;

public record Configuration(
    int intervalsCountToCheckLookahead,
    int intervalsCountToTriggerExtraction,
    int maximumComponentSize
) {
    final static public Configuration DEFAULT =
        new Configuration(24, 16, 100000);
}

File: ailist-core/src/main/scala/me/kosik/interwalled/ailist/core/model/Configuration.scala
package me.kosik.interwalled.ailist.core.model


final case class Configuration(
    intervalsCountToCheckLookahead: Int,
    intervalsCountToTriggerExtraction: Int,
    maximumComponentSize: Int
)

object Configuration {
    val DEFAULT: Configuration = Configuration(24, 16, 100000)

    def apply(): Configuration =
        Configuration(24, 16, 100000)
}

